frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.unit.spec.tsx

```diff
@@ -1,43 +1,67 @@
 import { checkNotNull } from "metabase/core/utils/types";
 import { getMetadata } from "metabase/selectors/metadata";
-import { SchemaId, TableId } from "metabase-types/api";
+import { InitialSyncStatus, Schema, Table } from "metabase-types/api";
 import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
 import { createMockState } from "metabase-types/store/mocks";
 import { createMockEntitiesState } from "__support__/store";
-import { renderWithProviders, screen } from "__support__/ui";
+import { renderWithProviders, screen, within } from "__support__/ui";
 import DataSelectorTablePicker from "./DataSelectorTablePicker";
 
 const TEST_TABLE = createMockTable();
 const TEST_DATABASE = createMockDatabase({ tables: [TEST_TABLE] });
```

Comment:
Since we created `TEST_DATABASE` inside `setup` this test now looks a bit disconnected.
https://github.com/metabase/metabase/blob/59bace24a5f8bfd2e43624371e6e5e418867af75/frontend/src/metabase/query_builder/components/DataSelector/DataSelectorTablePicker/DataSelectorTablePicker.unit.spec.tsx#L65-L73

You could see that we use the global `TEST_DATABASE` in the assertion, but it doesn't use the global `TEST_DATABASE` in the `setup()` function at all. Maybe there could be a way to pass `tables` without hard-coding this global `TEST_DATABASE`?

Another thing is that there are 2 variables named `TEST_DATABASE` in different context (global vs `setup` scope) and that could be confusing.

Reply:
Fixed